pkg/input/input.go

```diff
@@ -0,0 +1,108 @@
+package input
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+const (
+	// oktetoCaCertEnvVar is the environment variable that contains the CA certificate
+	oktetoCaCertEnvVar = "OKTETO_CA_CERT"
+)
+
+var (
+	// ErrInvalidInput is returned when the input is invalid
+	ErrInvalidInput = errors.New("invalid input")
+
+	// ErrInsufficientArguments is returned when the input arguments are insufficient
+	ErrInsufficientArguments = errors.New("insufficient arguments")
+)
+
+type Input struct {
+	Name      string
+	Namespace string
+	File      string
+	Deploy    bool
+	NoCache   bool
+	Variables []string
+	Timeout   string
+	Tests     string
+	LogLevel  string
+	CaCert    string
+}
+
+// NewInput parses the arguments and environment variables into an Input struct
+func NewInput(args []string, envVars map[string]string) (*Input, error) {
+	if len(args) < 9 {
+		return nil, fmt.Errorf("%w: %w", ErrInvalidInput, ErrInsufficientArguments)
+	}
+
+	input := &Input{
+		Name:      args[0],
+		Namespace: args[1],
+		File:      args[2],
+		Deploy:    loadBoolOrDefault(args[3], false),
+		NoCache:   loadBoolOrDefault(args[4], false),
+		Variables: parseVariables(args[5]),
+		Timeout:   args[6],
+		Tests:     args[7],
+		LogLevel:  args[8],
+		CaCert:    envVars[oktetoCaCertEnvVar],
+	}
+	input.Variables = append(input.Variables, getGithubEnvVars(envVars)...)
+	return input, nil
+}
+
+func loadBoolOrDefault(value string, defaultValue bool) bool {
+	if value == "" {
+		return defaultValue
+	}
+	trueValues := map[string]bool{
+		"true": true,
+		"t":    true,
+		"1":    true,
+		"yes":  true,
+		"y":    true,
+	}
```

Comment:
Nit: Would it make sense to have this at file level instead of generating this map every time `loadBoolOrDefault` is called?